io.home-assistant/thermostat.js

```diff
@@ -0,0 +1,156 @@
+        }
+        else {
+            throw new Error ('Sorry! Your device doesn\'t support checking humidity.');
+        }
+    }
+    // note: subscribe_ must NOT be async, or an ImplementationError will occur at runtime
+    subscribe_humidity() {
+        if (['binary_sensor', 'sensor'].includes(this.domain) && this.device_class === 'humidity') {
+            if (this.domain === 'sensor') {
+                let value = parseFloat(this.state.state);
+                return this._subscribeState(() => {
+                    return {state: undefined, value: value};
+                });
+            } else if (this.domain === 'binary_sensor') {
+                let state = this.deviceStateMapping[this.state.state];
+                return this._subscribeState(() => {
+                    return {state: state, value: undefined};
+                });
+            } else {
+                throw new Error (`Unexpected Home Assistant domain ${this.domain}`);
+            }
+        } else if (this.domain === 'climate') {
+            let value = parseFloat(this.state.current_humidity);
+            return this._subscribeState(() => {
+                return {state: undefined, value: value};
+            });
+        }
+        else {
+            throw new Error ('Sorry! Your device doesn\'t support checking humidity.');
+        }
+    }
+    async get_temperature() {
+        if (['binary_sensor', 'sensor'].includes(this.domain) && this.device_class === 'temperature') {
+            return this._get_state();
+        }
+        else if (this.domain === 'climate') {
+            let value = parseFloat(this.state.current_temperature);
+            return [{state: undefined, value: value}];
+        }
+        else {
+            throw new Error ('Sorry! Your device doesn\'t support checking temperature.');
+        }
+    }
+    // note: subscribe_ must NOT be async, or an ImplementationError will occur at runtime
+    subscribe_temperature() {
+        if (['binary_sensor', 'sensor'].includes(this.domain) && this.device_class === 'temperature') {
+            if (this.domain === 'sensor') {
+                let value = parseFloat(this.state.state);
+                return this._subscribeState(() => {
+                    return {state: undefined, value: value};
+                });
+            } else {
+                throw new Error (`Unexpected Home Assistant domain ${this.domain}`);
+            }
+        } else if (this.domain === 'climate') {
+            let value = parseFloat(this.state.current_temperature);
+            return this._subscribeState(() => {
+                return {state: undefined, value: value};
+            });
+        }
+        else {
+            throw new Error ('Sorry! Your device doesn\'t support checking temperature.');
+        }
+    }
+    
+    async get_hvac_state() {
+        if (this.domain === 'climate') {
+            let mode = this.state.hvac_mode;
+            let state = this.state.hvac_action;
```

Comment:
Are the enums compatible between nest and home assistant?

Reply:
HA has a few more enum values that I added in the manifest.tt. I don't have to change anything on the nest side for this right?

Reply:
For queries no, but nest also has a set_hvac_state that probably should throw for unsupported values.

Reply:
Threw unsupported values in latest commit